gyp/gyp_main.py

```diff
@@ -6,10 +6,38 @@
 
 import os
 import sys
+import subprocess
+
+# Below IsCygwin() function copied from pylib/gyp/common.py
+def IsCygwin():        
```

Comment:
Why is this defined twice, can't we put it into a utility somewhere? even its own file, however it's done in python

Reply:
As discussed at https://github.com/nodejs/node-gyp/pull/1817#pullrequestreview-260141661

Reply:
ok, if you're taking ownership of it. make sure that issue is created

Reply:
#1824